internals/overlord/servstate/manager_test.go

```diff
@@ -408,6 +414,70 @@ services:
 	s.stopTestServices(c)
 }
 
+func (s *S) TestReplanServicesWithWorkload(c *C) {
+	s.newServiceManager(c)
+	s.planAddLayer(c, testPlanLayer)
+
+	resetWorkloadsSectionExtension()
+	s.planAddLayer(c, `
+services:
+    test6:
+        override: replace
+        startup: enabled
+        command: /bin/test6
+        workload: default
+workloads:
+    default:
+        override: replace
+        user: nobody
+        group: nogroup
```

Comment:
Wouldn't these fail to start unless running the tests as root? IIRC, a non-root user can't start a process as another user/group.

Reply:
That is true, however, for this test specifically I do not start the services--only replan and test whether the services would be started after changing workload configuration.